Week01/Problem03/aster_03.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Jul 24 17:33:52 2017
+
+@author: Aster
+"""
+
+def ProjectEuler_problem3(N):
+    """
+    The prime factors of 13195 are 5, 7, 13 and 29.
+    What is the largest prime factor of the number 600851475143 ?
+    """
+    def find_two_factors(n):
```

Comment:
Defining functions inside other functions, while not against the rules of Python, might have unintended consequences...for instance, your function `find_two_factors()` *does not exist* outside of the function `ProjectEuler_problem3` because of the concept of *scope*.  In short, any variable or function defined within a function only exists within the surrounding function.  So, if you ever wanted to use this `find_two_factors` function again, you wouldn't be able to.  

Again, it's not against the rules, but it may not be what you want to do in the future.  Or it may be exactly what you want to do, depending on the context.